lib/matplotlib/cm.py

```diff
@@ -100,6 +100,12 @@ def register_cmap(name=None, cmap=None, data=None, lut=None):
             name = cmap.name
         except AttributeError:
             raise ValueError("Arguments must include a name or a Colormap")
+    if name in cmap_d:
+        msg = f"Trying to register the cmap {name} which already exists."
+        if name in globals():
```

Comment:
checking globals() looks awkward

Reply:
I think that if you want the check to be meaningful at all you need to hide cmap_d away from public use (or (ugh!) switch to it a dict subclass that prevents writes), otherwise people can just write into it...

Reply:
I don't think we have another listing of the built in color maps.

Reply:
then perhaps we should have (a private) one.

Reply:
Maybe assuming that noone edits the `datad` and `cmap_listed` dictionaries, 

```
alld = set([*plt.cm.datad.keys(), *plt.cm.cmaps_listed.keys()])
assert alld == plt.cm.cmap_d.keys() 
```